src/app/service/util.service.ts

```diff
@@ -1,9 +1,21 @@
 import { Injectable } from '@angular/core';
+import {AbstractControl} from "@angular/forms";
 
 @Injectable({
   providedIn: 'root'
 })
 export class UtilService {
 
   constructor() { }
+
+  emailPatternMatcher(control: AbstractControl): { [key: string]: any } | null {
+    const email: string = control.value;
```

Comment:
instead of using control.value here you can send control.value as a param

Reply:
If I am adding validator functions in formBuilder.group it shows error.
that's why i Added abstractControl

Reply:
ok can you move this to a form validations file so that it has exclusively functions for form validations? it will lead to a better understanding